gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapperTest.java

```diff
@@ -183,6 +205,90 @@ public void mapObservationToImmunizationWithUNKVaccineCode() {
         assertImmunizationWithDefaultVaccineCode(immunization);
     }
 
+    @Test
+    public void When_EhrCompositionWithNoPatConfidentialityCode_Expect_MetaFromConfidentialityServiceWithSecurity() {
+        final var metaWithSecurity = MetaFactory.getMetaFor(META_WITH_SECURITY, META_PROFILE);
+        when(
+            confidentialityService.createMetaAndAddSecurityIfConfidentialityCodesPresent(
+                eq("Immunization-1"),
+                confidentialityCodeCaptor.capture(),
+                confidentialityCodeCaptor.capture()
+            )
+        ).thenReturn(metaWithSecurity);
+        final var ehrExtract = unmarshallEhrExtract("immunization_with_ehrComposition_nopat_confidentiality_code.xml");
+
+        List<Immunization> immunizations = immunizationMapper.mapResources(
+            ehrExtract, getPatient(),
+            getEncounterList(), PRACTISE_CODE
+        );
+        var securityMeta = immunizations
+            .get(0)
+            .getMeta()
+            .getSecurity(NOPAT_URL_CODESYSTEM, NOPAT_CODE);
+
+        assertAll(
+            () -> {
+                assertThat(confidentialityCodeCaptor.getAllValues().get(0).isPresent())
+                    .isTrue();
```

Comment:
I think this can be shortened!

```suggestion
                assertThat(confidentialityCodeCaptor.getAllValues().get(1))
                    .isPresent();
```